libs/mrwebrtc/include/remote_audio_track_interop.h

```diff
@@ -52,10 +59,90 @@ mrsRemoteAudioTrackIsEnabled(mrsRemoteAudioTrackHandle track_handle) noexcept;
 MRS_API void MRS_CALL
-mrsRemoteAudioTrackOutputToDevice(mrsRemoteAudioTrackHandle track_handle, bool output) noexcept;
+mrsRemoteAudioTrackOutputToDevice(mrsRemoteAudioTrackHandle track_handle,
+                                  bool output) noexcept;
 
 /// Returns whether the track is output directly to the system audio device.
-MRS_API mrsBool MRS_CALL
-mrsRemoteAudioTrackIsOutputToDevice(mrsRemoteAudioTrackHandle track_handle) noexcept;
+MRS_API mrsBool MRS_CALL mrsRemoteAudioTrackIsOutputToDevice(
+    mrsRemoteAudioTrackHandle track_handle) noexcept;
+
+/// High level interface for consuming WebRTC audio tracks.
+///
+/// Enqueues audio frames for a remote audio track in an internal buffer as they
+/// arrive. Users should call |mrsAudioTrackReadBufferRead| to read samples from the buffer when needed.
+/// See also |mrsRemoteAudioTrackCreateReadBuffer|.
+using mrsAudioTrackReadBufferHandle = void*;
+
+/// Controls the padding behavior of |mrsAudioTrackReadBufferRead| on underrun.
+enum class mrsAudioTrackReadBufferPadBehavior : int32_t {
+
+  /// Do not pad the samples array.
+  kDoNotPad = 0,
+
+  /// Pad with zeros (silence).
+  kPadWithZero = 1,
+
+  /// Pad with a sine function.
+  /// Generates audible artifacts on underrun. Use for debugging.
+  kPadWithSine = 2,
+
+  kCount
+};
+
+
+/// Starts buffering the audio data from the remote track in an
+/// AudioTrackReadBuffer.
+///
+/// WebRTC audio tracks produce an audio frame every 10 ms.
+/// If you want the audio frames to be buffered (and optionally resampled)
+/// automatically, and you want the application to control when new audio data
+/// is read, create an AudioTrackReadBuffer using this function. If you want to
+/// process the audio frames as soon as they are received, without conversions,
+/// use |mrsRemoteAudioTrackRegisterFrameCallback instead|.
+MRS_API mrsResult MRS_CALL
+mrsRemoteAudioTrackCreateReadBuffer(mrsRemoteAudioTrackHandle track_handle,
+                                    mrsAudioTrackReadBufferHandle* bufferOut);
+
+
+/// Fill |data| with samples from the internal buffer.
+///
+/// This method reads the internal buffer starting from the oldest data.
+/// If the internal buffer is exhausted (underrun), |data|
+/// is padded according to the value of |padBehavior|.
+///
+/// This method should be called regularly to consume the audio data as it is
+/// received. Note that the internal buffer can overrun (and some frames can be
+/// dropped) if this is not called frequently enough.
+///
+/// |sample_rate|: Desired sample rate. Data in the buffer is resampled if this
+/// is different from the native track rate.
+///
+/// |num_channels|: Desired number of channels. Should be 1 or 2. Data in the
+/// buffer is split/averaged if this is different from the native track channels
+/// number.
+///
+/// |pad_behavior|: Controls how |data| is padded in case of underrun.
+///
+/// |data_out|: Will be filled with the samples read from the internal buffer.
```

Comment:
Let's be explicit that the buffer is allocated by the user (obvious from the signature) and how big the buffer needs to be (less obvious)

Reply:
Forcing the caller to pass the size might help users from making mistakes, even though it's redundant. It means we can give better error messages too.

Reply:
Fixed